src/utils.py

```diff
@@ -191,39 +185,80 @@ def get_sha256_response(sha256_url: str) -> requests.Response:
 
 
 def download_chain_spec(url, filename):
-    # Download file
-    file_response = requests.get(url, timeout=None)
-    if not CHAIN_SPEC_PATH.exists():
-        CHAIN_SPEC_PATH.mkdir(parents=True)
-    with open(Path(CHAIN_SPEC_PATH, filename), 'wb') as f:
-        f.write(file_response.content)
-    sp.run(['chown', '-R', f'{USER}:{USER}', CHAIN_SPEC_PATH], check=False)
+    if not c.CHAIN_SPEC_PATH.exists():
+        c.CHAIN_SPEC_PATH.mkdir(parents=True)
+    try:
+        download_file(url, Path(c.CHAIN_SPEC_PATH, filename))
+    except ValueError as e:
+        logger.error(f'Failed to download chain spec: {e}')
+        raise e
+
+
+def download_wasm_runtime(url):
+    if not url:
+        logger.debug('No wasm runtime url provided, skipping download')
+        return
+    filename = Path(url.split('/')[-1])
+    if not filename.name.endswith('.tar.gz') and not filename.suffix == '.wasm':
```

Comment:
What about other archive file suffixes like `.tar` or `.tar.zst`?

Not saying we need to support them in this PR, especially if you haven't seen any wasms being shared using any of them, but something to think about perhaps.